From e141adc070277444f0d8131c75e43e77e8101f32 Mon Sep 17 00:00:00 2001
From: Timothy Nikkel <tnikkel@gmail.com>
Date: Tue, 27 Oct 2026 20:35:37 +0000
Subject: [PATCH] Bug 2084560. Fix libyuv link failure on 32-bit x86
 non-Windows builds. r=ng

Since the update to 9456b062, libyuv implements the transpose functions for
32-bit x86 only in rotate_win.cc, on any OS and compiler. But libyuv.gyp sets
chromium_code, which pulls in filename_rules.gypi, and that excludes *_win.cc
on non-Windows. Upstream's GN and CMake builds compile these files everywhere
and rely on their #if guards, so we do the same.

Differential Revision: https://phabricator.services.mozilla.com/D335939
---
 ...include_win_sources_on_all_platforms.patch | 20 +++++++++++++++++++
 media/libyuv/libyuv/libyuv.gyp                | 10 ++++++++++
 media/libyuv/moz.yaml                         |  1 +
 3 files changed, 31 insertions(+)
 create mode 100644 media/libyuv/05_include_win_sources_on_all_platforms.patch

diff --git a/media/libyuv/05_include_win_sources_on_all_platforms.patch b/media/libyuv/05_include_win_sources_on_all_platforms.patch
new file mode 100644
index 0000000000000..6918296b53e6a
--- /dev/null
+++ b/media/libyuv/05_include_win_sources_on_all_platforms.patch
@@ -0,0 +1,20 @@
+diff --git a/libyuv.gyp b/libyuv.gyp
+--- a/libyuv.gyp
++++ b/libyuv.gyp
+@@ -148,6 +148,16 @@
+       'sources': [
+         '<@(libyuv_sources)',
+       ],
++      # Undo filename_rules.gypi's exclusion of *_win.cc on non-Windows. The
++      # files are guarded by #if, and rotate_win.cc is used for 32-bit x86 on
++      # any OS.
++      'target_conditions': [
++        ['OS != "win"', {
++          'sources/': [
++            ['include', '_win\\.cc$'],
++          ],
++        }],
++      ],
+     },
+     {
+       'target_name': 'libyuv_neon',
diff --git a/media/libyuv/libyuv/libyuv.gyp b/media/libyuv/libyuv/libyuv.gyp
index 4104659909afa..e528d240f0170 100644
--- a/media/libyuv/libyuv/libyuv.gyp
+++ b/media/libyuv/libyuv/libyuv.gyp
@@ -148,6 +148,16 @@
       'sources': [
         '<@(libyuv_sources)',
       ],
+      # Undo filename_rules.gypi's exclusion of *_win.cc on non-Windows. The
+      # files are guarded by #if, and rotate_win.cc is used for 32-bit x86 on
+      # any OS.
+      'target_conditions': [
+        ['OS != "win"', {
+          'sources/': [
+            ['include', '_win\\.cc$'],
+          ],
+        }],
+      ],
     },
     {
       'target_name': 'libyuv_neon',
diff --git a/media/libyuv/moz.yaml b/media/libyuv/moz.yaml
index 00c8e80c38223..b518626d90395 100644
--- a/media/libyuv/moz.yaml
+++ b/media/libyuv/moz.yaml
@@ -57,4 +57,5 @@ vendoring:
     - 02_update_gyp.patch
     - 03_add_neon64_and_sve_gyp_targets.patch
     - 04_disable_sme.patch
+    - 05_include_win_sources_on_all_platforms.patch
 
